src/components/RepoForm/RepoForm.js

```diff
@@ -90,7 +103,7 @@ export default class RepoForm extends Component {
           </label>
         </div>
 
-        <button className={(loading ? 'disabled ' : '') + 'btn btn-block btn-primary'}>
+        <button disabled={this.state.invalidFields} className={(loading ? 'disabled ' : '') + 'btn btn-block btn-primary'}>
```

Comment:
So notice how I'm using disabled in the class for loading. If you're planning to add the disabled attribute move my loading disabled with it so that its disabled={ invalidFields || loading } (Also grab the invalidFields using the ES6 notation. const { invalidFields } = this.state; Similar to what I did at the beginning of the render method for props.